### PR TITLE
Feature/ar 147 advanced filter

### PR DESCRIPTION
**Pull request by AlexeyDerepa**

### CODE REVIEW

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/Alias.cs [hunk 1]
```diff
@@ -2,12 +2,11 @@
 
 namespace AnimalRescue.DataAccess.Mongodb.QueryBuilders
 {
-    internal class Alias
+    public class Alias
     {
         public string PropertyName { get; set; }
         public string DataBasePropertyName { get; set; }
         public string AliasePropertyName { get; set; }
```
Comment (dmytrokorotkyi) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/Alias.cs]: Please rename AliasePropertyName to AliasPropertyName

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs [hunk 2]
```diff
@@ -27,7 +27,7 @@ public AliasStore()
 
         public Alias GetAlias<T>(string aliasePropertyName)
```
Comment (dmytrokorotkyi) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs]: Please rename aliasePropertyName to aliasPropertyName

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs [hunk 3]
```diff
@@ -27,7 +27,7 @@ public AliasStore()
...
         {
-            Type type = typeof(T);            
+            Type type = typeof(T);
 
             return GetAlias(type, aliasePropertyName);
```
Comment (dmytrokorotkyi) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs]: Please rename aliasePropertyName to aliasPropertyName

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs [hunk 4]
```diff
@@ -46,6 +53,23 @@ public Alias GetAlias(Type type, string aliasePropertyName)
                 .FirstOrDefault(alias => IsEqualNames(alias, aliasePropertyName));
         }
 
+        private Alias FindNestedAlias(string aliasePropertyName)
```
Comment (dmytrokorotkyi) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs]: Please rename aliasePropertyName to aliasPropertyName

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/IAliasStore.cs [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 namespace AnimalRescue.DataAccess.Mongodb.QueryBuilders
 {
-    internal interface IAliasStore
+    public interface IAliasStore
     {
         Alias GetAlias<T>(string aliasePropertyName);
```
Comment (dmytrokorotkyi) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/IAliasStore.cs]: Please rename aliasePropertyName to aliasPropertyName

Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/IAliasStore.cs]: done

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs (hunk 4 above)
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs]: done

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs (hunk 3 above)
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs]: done

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs (hunk 2 above)
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/AliasStore.cs]: done

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/Alias.cs (hunk 1 above)
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/QueryBuilders/Alias.cs]: done

**Pull request merged**